neo/test/actor/RequestRouterTest.scala

```diff
@@ -1,30 +1,221 @@
+      |      { "field":"text","relation":"contains","values":["zika","virus"]}
+      |    ],
+      |    "group":{
+      |      "by":[
+      |        { "field":"create_at",
+      |          "apply":{"name":"interval","args":{"unit":"hour"}},
+      |          "as":"hour"}
+      |      ],
+      |      "aggregate":[
+      |        { "field":"*",
+      |          "apply":{"name":"count"},
+      |          "as":"count"}
+      |      ]
+      |    }
+      |  }],
+      |  "option": {
+      |    "sliceMillis": 2000
+      |  },
+      |  "transform":{
+      |    "wrap":{
+      |      "key": "batch"
+      |    }
+      |  }
+      |}
+      |""
+    """.stripMargin)
+
+  val nonSlicingRequestWithNoTransform = Json.parse(
+    """
+      |{ "dataset":"twitter",
+      |  "filter":[
+      |    {"field":"geo_tag.cityID","relation":"in","values":[1,2,3,4]},
+      |    {"field":"create_at","relation":"inRange","values":["1970-01-01T00:00:00.000Z","1970-01-01T00:00:02.000Z"]},
+      |    {"field":"text","relation":"contains","values":["zika","virus"]}
+      |  ],
+      |  "select":{
+      |    "order":["-create_at"],
+      |    "limit":10,
+      |    "offset":0,
+      |    "field":["create_at","id","user.id"]
+      |  }
+      |}
+    """.stripMargin)
+
+  val slicingRequestWithNoTransform = Json.parse(
+    """
+      |{ "batch": [
+      |  { "dataset":"twitter",
+      |    "filter":[
+      |      { "field":"geo_tag.cityID","relation":"in","values":[1,2,3,4]},
+      |      { "field":"create_at","relation":"inRange","values":["1970-01-01T00:00:00.000Z","1970-01-01T00:00:02.000Z"]},
+      |      { "field":"text","relation":"contains","values":["zika","virus"]}
+      |    ],
+      |    "group":{
+      |      "by":[
+      |        { "field":"create_at",
+      |          "apply":{"name":"interval","args":{"unit":"hour"}},
+      |          "as":"hour"}
+      |      ],
+      |      "aggregate":[
+      |        { "field":"*",
+      |          "apply":{"name":"count"},
+      |          "as":"count"}
+      |      ]
+      |    }
+      |  }],
+      |  "option": {
+      |    "sliceMillis": 2000
+      |  }
+      |}
+      |""
+    """.stripMargin)
+
   "RequestRouter" should {
-    "XXXX" in {
-      // TODO need to be implemented
+    "parse transform and pass transform to berry" in {
+
+      val berryResult = Json.obj("result" -> "result example")
+      val frontEnd = new TestProbe(system)
+      val clientProps = Props(new Actor {
+        override def receive: Receive = {
+          case berryRequest: Request =>
+            frontEnd.ref ! berryRequest.postTransform.transform(berryResult)
+          case _ =>
+            throw new IllegalArgumentException("Message type sent to berry is not correct.")
+        }
+      })
+      val router = system.actorOf(RequestRouter.props(frontEnd.ref, clientProps, Config.Default))
```

Comment:
since the `clientProps` will create two different actors inside, then the `frontEnd` should receive two kinds of msgs from different actors, not the same one, then if all msgs are complete, and different msgs are coming from two different senders, it serves the purpose of testing the `router`.